app/models/concerns/hyku_addons/account_behavior.rb

```diff
@@ -74,8 +75,27 @@ def switch_host!(cname)
         Rails.application.routes.default_url_options[:host] = cname
         Hyrax::Engine.routes.default_url_options[:host] = cname
       end
+
+      def setup_tenant_cache
+        Rails.application.config.action_controller.perform_caching = api_cache?
+        ActionController::Base.perform_caching = api_cache?
+
+        Rails.application.config.cache_store = :redis_cache_store, { url: Redis.current.id } if api_cache?
+        Rails.cache = ActiveSupport::Cache.lookup_store(Rails.application.config.cache_store)
+      end
```

Comment:
Now that we have boiled down this method and seen exactly what is necessary, can I suggest that instead of the mutliple calls to `api_cache?`, we do something like: 

      def setup_tenant_cache
        if api_cache?
          Rails.application.config.action_controller.perform_caching = true
          ActionController::Base.perform_caching = true
          Rails.application.config.cache_store = :redis_cache_store, { url: Redis.current.id } 
        end

        Rails.cache = ActiveSupport::Cache.lookup_store(Rails.application.config.cache_store)
      end

This then makes it clear that if the caching is enabled, then we are altering the default settings for the tenant.